src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/container_uninitialized_ref.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+use Symfony\Component\DependencyInjection\Argument\RewindableGenerator;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+use Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag;
+
+/**
+ * Symfony_DI_PhpDumper_Test_Uninitialized_Reference.
+ *
+ * This class has been auto-generated
+ * by the Symfony Dependency Injection Component.
+ *
+ * @final since Symfony 3.3
+ */
+class Symfony_DI_PhpDumper_Test_Uninitialized_Reference extends Container
+{
+    private $parameters;
+    private $targetDirs = array();
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
+    {
+        $this->services = array();
+        $this->methodMap = array(
+            'bar' => 'getBarService',
+        );
+
+        $this->aliases = array();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function compile()
+    {
+        throw new LogicException('You cannot compile a dumped container that was already compiled.');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCompiled()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isFrozen()
+    {
+        @trigger_error(sprintf('The %s() method is deprecated since version 3.3 and will be removed in 4.0. Use the isCompiled() method instead.', __METHOD__), E_USER_DEPRECATED);
+
+        return true;
+    }
+
+    /**
+     * Gets the public 'bar' shared service.
+     *
+     * @return \stdClass
+     */
+    protected function getBarService()
+    {
+        $this->services['bar'] = $instance = new \stdClass();
+
+        $instance->ref = ${($_ = isset($this->services['foo']) ? $this->services['foo'] : null) && false ?: '_'};
+        $instance->lambda = function () {
+            return ${($_ = isset($this->services['foo']) ? $this->services['foo'] : null) && false ?: '_'};
+        };
+        $instance->iter = new RewindableGenerator(function () {
+            if (isset($this->services['foo'])) {
+                yield 'foo' => ${($_ = isset($this->services['foo']) ? $this->services['foo'] : null) && false ?: '_'};
```

Comment:
can't we optimize this, as it repeats the wrapping condition ? Or is it not worth it ?

Reply:
would add complexity, not worth it imho